src/ARC/ARC.fs

```diff
@@ -94,6 +94,64 @@ type ARC =
 
     // to-do: function that returns read contracts based on a list of paths.
     // the list of paths is used to create a filesystem tree
-    static member createReadContracts (paths : string array) : Contract array =
-        let fileTree = FileSystemTree.fromFilePaths paths
-        raise (System.NotImplementedException())
\ No newline at end of file
+    static member createReadContracts (filePaths : string array) : Contract array =
+        let xlsxReadContractFromPath (path: string) = Contract.createRead(path, DTOType.Spreadsheet)
+        let fs = FileSystem.fromFilePaths filePaths
+        let xlsxFileNames = [|Path.AssayFileName; Path.StudyFileName; Path.InvestigationFileName|]
```

Comment:
Yeah if i don't do this, i more or less need to implement the whole "Identifier" logic from ISA.Spreadsheet again. I just parse everything and check identifiers afterwards. This way we have this logic maintainable.

Otherwise this was my idea for study:

```fsharp
let expectedStudyPathFromStudyIdentifier (studyIdentifier) =
    FileSystem.Path.combineMany[|Path.ARCtrl.StudiesFolderName; studyIdentifier; Path.ARCtrl.StudyFileName|]

let tryFindStudy (contracts: Contract []) (studyRegisteredIdent: string) =
    contracts |> Array.tryPick (fun c ->
        let expectedPath = expectedStudyPathFromStudyIdentifier studyRegisteredIdent
        match c with
        | {Operation = READ; DTOType = Some DTOType.ISA_Study; DTO = Some (DTO.Spreadsheet fsworkbook); Path = p} when p = expectedPath ->
            Some (p, fsworkbook)
        | _ -> None
    )
```

But if identifier and foldername are not the same anymore this will simply not work anymore.

Reply:
IMO beautifully solved with:

```fsharp
// Assumptions: 
// 1. All ISA .xlsx files MUST have the same names, either "isa.investigation.xlsx", "isa.assay.xlsx" or "isa.study.xlsx"
// 2. All ISA .xlsx files MUST be placed at the correct place. 
//      - isa.investigation.xlsx: MUST be in root
//      - isa.assay.xlsx: MUST be in root/assays/ANY_ASSAY_NAME/
//      - isa.study.xlsx: MUST be in root/studies/ANY_STUDY_NAME/
// 3. We ignore all other .xlsx files.
/// Tries to create READ contract with DTOType = ISA_Assay, ISA_Study or ISA_Investigation from given path relative to ARC root.
let tryISAReadContractFromPath (path: string) = 
    let split = FileSystem.Path.split path
    match split with
    | [|InvestigationFileName|] -> 
        Some <| Contract.createRead(path, DTOType.ISA_Investigation) 
    | [|AssaysFolderName; anyAssayName; AssayFileName|] ->
        Some <| Contract.createRead(path, DTOType.ISA_Assay) 
    | [|StudiesFolderName; anyStudyName; StudyFileName|] ->
        Some <| Contract.createRead(path, DTOType.ISA_Study) 
    | anyElse -> 
        None
```